### PR TITLE
[Jest] Migrate `scroll_zoom.test.js`

### PR DESCRIPTION
**Pull request by wipfli**
- Move and rename
- Run jscodeshift
- Run custom codemods and lint
- Manual edits

## Launch Checklist

This is still work in progress.

 - [x] Confirm **your changes do not include backports from Mapbox projects** (unless with compliant license) - if you are not sure about this, please ask!
 - [ ] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [ ] Write tests for all new functionality.
 - [ ] Document any changes to public APIs.
 - [ ] Post benchmark scores.
 - [ ] Manually test the debug page.
 - [ ] Suggest a changelog category: bug/feature/docs/etc. or "skip changelog".
 - [ ] Add an entry inside this element for inclusion in the `maplibre-gl-js` changelog: `<changelog></changelog>`.

### CODE REVIEW

**Comment by wipfli** [comment]
This unit test used sinon mixed with some home-made artificial timing. Should I refactor the whole test and do everything with Jest's fake timers?﻿

**Comment by HarelM** [comment]
I vote for yes. If we can improve the while refactoring we are doing the right thing.

**Comment by wipfli** [comment]
Ready for review @HarelM

File: src/ui/handler/scroll_zoom.test.ts [hunk 1]
```diff
@@ -229,16 +264,12 @@ test('ScrollZoomHandler', (t) => {
...
         }
 
-        clock.tick(200);
+        jest.advanceTimersByTime(200);
         map._renderTaskQueue.run();
 
-        t.equal(startCount, 1);
-        t.equal(endCount, 1);
-
-        clock.restore();
```
Comment (wipfli) [inline src/ui/handler/scroll_zoom.test.ts:238]: `jest.useRealTimers()` can be added if needed

**Review by HarelM — APPROVED** [review]
Nicely done!

**Comment by wipfli** [comment]
Thanks :)

**Pull request merged**